competency-explorer/src/thunks/load.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  loadEngCompetencies,
+  loadManagerCompetencies,
+} from "../helpers/loadFile";
+import { parseFile } from "../helpers/parseFile";
+import {
+  createLevelAction,
+  createCategoryAction,
+  createCompetencyAction,
+  createFilterAction,
+} from "../reducers";
+import type { Dispatch } from "react";
+
+
+const parseFileIntoState = (fileContents: string, dispatch: Dispatch<any>, hideByDefault?: boolean) => {
+  const parsed = parseFile(fileContents);
+
+  dispatch(createLevelAction(parsed.levels));
+  dispatch(createCategoryAction(parsed.categories));
+  dispatch(createCompetencyAction(parsed.competencies));
+
+  if (!hideByDefault) { return; }
+  for (let l of parsed.levels) {
+    dispatch(createFilterAction(l, 'HIDE_LEVEL'));
```

Comment:
Would it make sense to use the `HIDE_LEVEL` constant here in place of the raw `'HIDE_LEVEL'` string? OR wrap the action creator in a helper, something like `createHideLevelFilterAction`?

Reply:
typescript provides the same safety here as the constant itself (and auto-completes easier), so I opted for this approach